serde/src/java/org/apache/hadoop/hive/serde2/ColumnProjectionUtils.java

```diff
@@ -234,7 +235,7 @@ private static void appendReadColumnNames(Configuration conf, List<String> cols)
       if (first) {
         first = false;
       } else {
-        result.append(',');
+        result.append(conf.get(serdeConstants.COLUMN_NAME_DELIMITER, String.valueOf(SerDeUtils.COMMA)));
```

Comment:
```java
    boolean first = old.isEmpty();
    for(String col: cols) {
      if (first) {
        first = false;
      } else {
        result.append(conf.get(serdeConstants.COLUMN_NAME_DELIMITER, String.valueOf(SerDeUtils.COMMA)));
      }
      result.append(col);
    }
```

Can this just be simplified to something like:

```java
String delim = conf.get(serdeConstants.COLUMN_NAME_DELIMITER, String.valueOf(SerDeUtils.COMMA))
String xxx = String.join(delim, cols);
```

Reply:
Makes sense, did some refactoring based on your suggestion